### ISSUE TITLE
put overlay into display margin?

### ISSUE THREAD

**Issue by QuarterHeaven** [Artawower/blamer.el#68]
I'm using visual-fill-column (or [visual-line-mode](https://www.gnu.org/software/emacs/manual/html_node/emacs/Visual-Line-Mode.html)) to center the text and auto truncate lines, and met some truncate issues (like line 33 and 35). 

But there is a large blank space at the left and right of the lines called [display margins](https://www.gnu.org/software/emacs/manual/html_node/elisp/Display-Margins.html). Can we add an option to put the blamer overlays into the margins? I think it will be much more beautiful, and solve the truncate issues.

![image](https://github.com/Artawower/blamer.el/assets/41909950/c61dfc2a-1686-4020-80c9-679ea5bfb6fb)

**Comment by Artawower** [comment]
Yep, it should not be difficult. But I am not very familiar with margins. Could you please test this branch: https://github.com/Artawower/blamer.el/tree/feature/margin-overlays
To use margin overlay, just set `(setq blamer-type 'margin-overlay)`.

**Comment by QuarterHeaven** [comment]
> Yep, it should not be difficult. But I am not very familiar with margins. Could you please test this branch: https://github.com/Artawower/blamer.el/tree/feature/margin-overlays To use margin overlay, just set `(setq blamer-type 'margin-overlay)`.

It's amazing! But there are still some annoying truncates. Can we disable the truncation while using margin-overlay?

more intuitively, the full message is:

![image](https://github.com/Artawower/blamer.el/assets/41909950/761fd233-d711-4b72-8f30-721fcc7247fa)

but it still truncated while the code is long:

![image](https://github.com/Artawower/blamer.el/assets/41909950/60c682cb-2498-41fe-a90c-ca07bcb0ae07)

It's unnecessary in my thought.

**Comment by Artawower** [comment]
Agreed, check the last commit from the branch. You can disable the initial indentation by setting `(setq blamer-entire-formatter "%s")`.

**Comment by QuarterHeaven** [comment]
> Agreed, check the last commit from the branch. You can disable the initial indentation by setting `(setq blamer-entire-formatter "%s")`.

Great! It works perfectly as I expected. Thanks!

**Comment by igorcafe** [comment]
Same problem here with visual-line-mode

![image](https://github.com/Artawower/blamer.el/assets/85039990/6aa01ba7-8b90-4e9a-bb73-141f3040233e)
